topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -61,23 +74,51 @@
     /*test*/ ZkTopicWatcher topicWatcher;
     /*test*/ PrometheusMeterRegistry metricsRegistry;
     K8sTopicWatcher watcher;
-    /** The id of the periodic reconciliation timer. This is null during a periodic reconciliation. */
+    /**
+     * The id of the periodic reconciliation timer. This is null during a periodic reconciliation.
+     */
     private volatile Long timerId;
     private volatile boolean stopped = false;
     private Zk zk;
     private volatile HttpServer healthServer;
 
+    /**
+     * @param kubeClient kubernetes client
+     * @param config     TopicOperator config
+     */
     public Session(KubernetesClient kubeClient, Config config) {
+        this(kubeClient, config, null, null, new TopicOperatorState());
+    }
+
+    /**
+     * Constructor is package private for testing
+     *
+     * @param kubeClient         kubernetes client
+     * @param config             TopicOperator config
+     * @param topicStoreCreator  indirection to allow unit testing of start-up sequence with mock topicstore creation
+     * @param zkClientCreator    indirection to allow unit testing of start-up sequence with mock ZK client creation
+     * @param topicOperatorState used to communicate liveness/readiness to health server from things that can potentially fail
+     */
+    Session(KubernetesClient kubeClient, Config config, BiFunction<Zk, Config, TopicStore> topicStoreCreator, BiFunction<Vertx, Config, Future<Zk>> zkClientCreator, TopicOperatorState topicOperatorState) {
```

Comment:
So this is the new improved package private constructor for unit testing so that I can swap out creation of the Zk that uses a static factory method (but in a different thread, where Mockito's mockStatic capacity can't reach it), and the creation of the TopicOperatorStore that is either created via a static factory or a direct instantiation.